draft-js-inline-toolbar-plugin/src/components/Toolbar/index.js

```diff
@@ -45,23 +61,126 @@ export default class Toolbar extends React.Component {
+      // +-----------------------------------------------+
+      if (fromParentBeginning < toolbarHalfWidth) {
+        metrics.ruleValue = toolbarHalfWidth;
+        metrics.shift = 'left';
+      } else if (beforeParentEnd < toolbarHalfWidth) {
+        // the same, but relative to the parent end
+        // +-----------------------------------------------+
+        // |                                 vvv toolbar   |
+        // |                            +---------------+  |
+        // |                            +---------------+  |
+        // |                                               |
+        // |                                      +--+     |
+        // |                             selection ^^      |
+        // |                                               |
+        // +-----------------------------------------------+
+        metrics.ruleValue = -toolbarHalfWidth;
+        metrics.rule = 'right';
+        metrics.shift = 'right';
+      } else {
+        // selection somewhere in the middle within the parent and there is a
+        // free place for toolbar
+        metrics.ruleValue = (selectionRect.left - relativeRect.left)
+          + (selectionRect.width / 2);
+      }
+
+      // ok, why we have to use external style elements here? Because, we are
+      // moving pointer position each time and it can be set via `before`,
+      // `after` css-pseudo classes
+      const pointerClassName = 'draft-js-inline-toolbar-pointer';
+      const styleId = 'draft-js-inline-toolbar-pointer-pseudo';
+
+      if (this.toolbar.classList.contains(pointerClassName)) {
+        this.toolbar.classList.remove(pointerClassName);
+      }
+      // We are recreating style element on every toolbar appearance. It is
+      // better for us to use individual style element rather than join new
+      // styles to some of the existent elements, because we can easily remove
+      // it next time
+      if (document.getElementById(`${styleId}-${this.state.nonce}`)) {
+        document.head.removeChild(
+          document.getElementById(`${styleId}-${this.state.nonce}`)
+        );
+      }
+
+      if (typeof metrics.shift === 'string') {
+        const styleEl = document.createElement('style');
+        styleEl.setAttribute('id', `${styleId}-${this.state.nonce}`);
+        document.head.appendChild(styleEl);
+        this.toolbar.classList.add(pointerClassName);
+
+        if (metrics.shift === 'left') {
+          metrics.shiftValue = (toolbarHalfWidth - metrics.ruleValue) +
+            fromParentBeginning;
+        } else {
+          metrics.shiftValue = this.toolbar.offsetWidth -
+            (relativeRect.right - (selectionRect.right - (selectionRect.width / 2)));
+        }
+
+        styleEl.sheet.insertRule(`.${pointerClassName}::after { left: ${metrics.shiftValue}px}`, 0);
+        styleEl.sheet.insertRule(`.${pointerClassName}::before { left: ${metrics.shiftValue}px}`, 0);
```

Comment:
I'm very unclear why we need to manipulate the DOM and inject a stylesheet. Surely all the positioning logic can live inside `getStyle`

Reply:
@juliankrispel, seems like I found a solution. We can create react tag for our styles.